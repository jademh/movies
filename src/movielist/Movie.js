import React from 'react';
import PropTypes from 'prop-types';

export const POSTER_PATH = 'http://image.tmdb.org/t/p/w300';

export default function Movie(props) {
  const {
    title,
    releaseDate,
    posterPath,
  } = props;

  let movieTitle = title;
  if(releaseDate) {
    movieTitle += ` (${releaseDate.slice(0,4)})`;
  }

  if (posterPath === null) {
    return (
      <div data-testid="movie">
        <span data-testid="movie-title">{movieTitle}</span>
      </div>
    )
  }
  return (
    <div data-testid="movie">
      <img
        data-testid="movie-poster"
        src={`${POSTER_PATH}${posterPath}`}
        alt={movieTitle}
      />
    </div>
  )
}

Movie.propTypes = {
    title: PropTypes.string.isRequired,
    releaseDate: PropTypes.string,
    posterPath: PropTypes.string,
};
