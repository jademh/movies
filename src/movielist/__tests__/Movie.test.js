import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Movie, { POSTER_PATH } from '../Movie';

afterEach(() => {
  cleanup();
  console.error.mockClear();
});

console.error = jest.fn();

const movie = {
  title: 'Hackers',
  releaseDate: '1995-09-15',
  posterPath: '/image.jpg'
};

const movieWithNoPoster = {
  title: 'Hackers',
  releaseDate: '1995-09-15',
  posterPath: null
};

const movieWithNoReleaseDate = {
  title: 'Hackers',
  releaseDate: '',
  posterPath: '/image.jpg'
};

test('<Movie />', () => {
  render(<Movie />);
  expect(console.error).toHaveBeenCalled();
});

test('<Movie /> with poster and release date', () => {
  const { getByTestId, queryByTestId, container} = render(
      <Movie title={movie.title} releaseDate={movie.releaseDate} posterPath={movie.posterPath} />
  );
  expect(console.error).not.toHaveBeenCalled();
  expect(queryByTestId('movie-title')).toBe(null);
  expect(getByTestId('movie-poster').src).toBe(`${POSTER_PATH}${movie.posterPath}`);
  expect(getByTestId('movie-poster').getAttribute('alt')).toBe('Hackers (1995)');
  expect(container.firstChild).toMatchSnapshot();
});

test('<Movie /> with no poster', () => {
  const { getByTestId, queryByTestId, container } = render(
      <Movie title={movieWithNoPoster.title} releaseDate={movieWithNoPoster.releaseDate} posterPath={movieWithNoPoster.posterPath} />
  );
  expect(console.error).not.toHaveBeenCalled();
  expect(queryByTestId('movie-poster')).toBe(null);
  expect(getByTestId('movie-title').textContent).toBe('Hackers (1995)');
  expect(container.firstChild).toMatchSnapshot();
});

test('<Movie /> with no release date', () => {
  const { getByTestId, queryByTestId, container } = render(
      <Movie title={movieWithNoReleaseDate.title} releaseDate={movieWithNoReleaseDate.releaseDate} posterPath={movieWithNoReleaseDate.posterPath} />
  );
  expect(console.error).not.toHaveBeenCalled();
  expect(getByTestId('movie-poster').getAttribute('alt')).toBe('Hackers');
  expect(container.firstChild).toMatchSnapshot();
});