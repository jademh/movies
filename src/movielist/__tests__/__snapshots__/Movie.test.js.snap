// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movie /> with no poster 1`] = `
<div
  data-testid="movie"
>
  <span
    data-testid="movie-title"
  >
    Hackers (1995)
  </span>
</div>
`;

exports[`<Movie /> with no release date 1`] = `
<div
  data-testid="movie"
>
  <img
    alt="Hackers"
    data-testid="movie-poster"
    src="http://image.tmdb.org/t/p/w185/image.jpg"
  />
</div>
`;

exports[`<Movie /> with poster and release date 1`] = `
<div
  data-testid="movie"
>
  <img
    alt="Hackers (1995)"
    data-testid="movie-poster"
    src="http://image.tmdb.org/t/p/w185/image.jpg"
  />
</div>
`;
